src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -281,11 +299,17 @@ Terminal::UpdateSelectionParams Terminal::ConvertKeyEventToUpdateSelectionParams
     return std::nullopt;
 }
 
-void Terminal::UpdateSelection(SelectionDirection direction, SelectionExpansion mode)
+// Method Description:
+// - updates the selection endpoints based on a direction and expansion mode. Primarily used for keyboard selection.
+// Arguments:
+// - direction: the direction to move the selection endpoint in
+// - mode: the type of movement to be performed (i.e. move by word)
+// - (optional) moveSelectionEnd: used by mark mode. Can be used to explicitly move a specific endpoint. If true, move the "end" endpoint. Otherwise, move the start endpoint.
+void Terminal::UpdateSelection(SelectionDirection direction, SelectionExpansion mode, std::optional<bool> moveSelectionEnd)
 {
     // 1. Figure out which endpoint to update
     // One of the endpoints is the pivot, signifying that the other endpoint is the one we want to move.
-    const auto movingEnd{ _selection->start == _selection->pivot };
+    const auto movingEnd{ (_markMode && moveSelectionEnd.has_value()) ? *moveSelectionEnd : _selection->start == _selection->pivot };

```

Comment:
For some reason I'm having a hard time parsing this `moveSelectionEnd` logic here. So, there's three cases:
* `nullopt`: not in mark mode. use `_selection->start == _selection->pivot`. We'll later just move the end anchor of the selection.
* `true`: in mark mode. Shift was pressed. we're just going to move the end anchor?
* `false`: in mark mode. We're going to move the start. But then later we're in mark mode, so we're just going to move the end to where the start is?

idk something about this just felt weird.

Reply:
Ah, let me try and explain it this way...
- `nullopt` <-- we're just modifying an existing selection with the keyboard. So like, you created a selection with a mouse, and now you're just using <kbd>shift+arrows</kbd> to update it. Codeflow requires you to pivot around the "start" if you cross the endpoints.
- `true` <-- you are _exclusively_ only moving the "end". This is like when you're creating a selection in a text editor. "start" is anchored to the cursor, and "end" is moved; thus creating a selection.
- `false` <-- you are moving _both_ "start" and "end" together. This is when you're just moving the cursor in a text editor.

Reply:
that's a good description - mind adding that in the comments? That would make this clearer at least, to know what it is trying to do.